dali/server/daldap.cpp

```diff
@@ -105,32 +105,35 @@ class CDaliLdapConnection: implements IDaliLdapConnection, public CInterface
     {
         if (!ldapsecurity||((getLDAPflags()&DLF_ENABLED)==0))
             return SecAccess_Full;
-        StringBuffer username;
-        StringBuffer password;
-        if (udesc)
-        {
-            udesc->getUserName(username);
-            udesc->getPassword(password);
-        }
-        else
-        {
-            DBGLOG("NULL UserDescriptor in daldap.cpp getPermissions('%s')",key ? key : "NULL");
-        }
-
-        if (0 == username.length())
-        {
-            username.append(filesdefaultuser);
-            decrypt(password, filesdefaultpassword);
-            OWARNLOG("Missing credentials, injecting deprecated filesdefaultuser for request %s %s", nullText(key), nullText(obj));
-        }
-
-        Owned<ISecUser> user = ldapsecurity->createUser(username);
-        user->setAuthenticateStatus(AS_AUTHENTICATED);
 
         bool filescope = key && stricmp(key,"Scope")==0;
         bool wuscope = key && stricmp(key,"workunit")==0;
 
         if (filescope || wuscope) {
+
+            StringBuffer username;
+            StringBuffer password;
+            if (udesc)
+            {
+                udesc->getUserName(username);
+                udesc->getPassword(password);
+            }
+            else
+            {
+                DBGLOG("NULL UserDescriptor in daldap.cpp getPermissions('%s')",key ? key : "NULL");
+            }
+
+            if (0 == username.length())
+            {
+                username.append(filesdefaultuser);
+                decrypt(password, filesdefaultpassword);
+                OWARNLOG("Missing credentials, injecting deprecated filesdefaultuser for request %s %s", nullText(key), nullText(obj));
+                printStackReport();
```

Comment:
how much extra logging is this going to reproduce, if very verbose, there may be concerns re. the cost of logging e.g. on Azure Log Analytics in cloud.

the helper (logNullUser) is supposed to mitigate against that, by not logging every call, instead logging only if nothing logged in previous 60s.

logNullUser currently takes a IUserDescriptor, but it could be refactored into 2 helper methods, where IUserDescriptor methods calls one that takes a const char *, which can then be used here by CDaliLdapConnection::getPermissions.